pyodide-build/pyodide_build/xbuildenv.py

```diff
@@ -0,0 +1,332 @@
+import json
+import shutil
+import subprocess
+from pathlib import Path
+from tempfile import NamedTemporaryFile
+
+import requests
+from pyodide_lock import PyodideLockSpec
+
+from . import build_env
+from .create_pypi_index import create_pypi_index
+from .logger import logger
+
+XBUILDENV_URL = "https://github.com/pyodide/pyodide/releases/download/{version}/xbuildenv-{version}.tar.bz2"
+CDN_BASE = "https://cdn.jsdelivr.net/pyodide/v{version}/full/"
+
+
+class CrossBuildEnvManager:
+    """
+    Manager for the cross-build environment.
+    """
+
+    def __init__(self, env_dir: str | Path) -> None:
+        self.env_dir = Path(env_dir).resolve()
+
+        try:
+            self.env_dir.mkdir(parents=True, exist_ok=True)
+        except Exception as e:
+            raise ValueError(
+                f"Failed to create cross-build environment at {self.env_dir}"
+            ) from e
```

Comment:
Note that if you raise an exception in an `except` block, it is automatically chained to the original exception so it would be the same to do:
```py
except Exception:
  raise ValueError(...)
```